mir_perception/mir_perception_utils/ros/src/object_utils_ros.cpp

```diff
@@ -70,10 +72,38 @@ PointCloud object::estimatePose(const PointCloud::Ptr &xyz_input_cloud, geometry
     pcl::getMinMax3D(*xyz_input_cloud, min_pt, max_pt);
     double limit_min = min_pt.z + passthrough_lim_min_offset;
     double limit_max = max_pt.z + passthrough_lim_max_offset;
-
-    pass_through.setFilterLimits(limit_min, limit_max);
-    pass_through.setInputCloud(xyz_input_cloud);
-    pass_through.filter(filtered_cloud);
+    if (obj_category == "cavity")
+    {
+      // print
+      // std::cout << "[CAVITYYYYYYYYYYYY] min_pt.z: " << min_pt.z << std::endl;
+      // std::cout << "[CAVITYYYYYYYYYYYY] max_pt.z: " << max_pt.z << std::endl;
+      limit_max = max_pt.z - 0.015; // TODO: make 0.01 as a dynamic configurable parameter
```

Comment:
a few TODOs in this function! Some hard-coded numbers to be parameterized, and object names to be removed